plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/panel/GitPanelPresenter.java

```diff
@@ -0,0 +1,360 @@
+    this.locale = locale;
+
+    this.view.setDelegate(this);
+    this.view.setChangesPanelView(this.changesPanelPresenter.getView());
+
+    if (partStack == null || !partStack.containsPart(this)) {
+      workspaceAgent.openPart(this, PartStackType.NAVIGATION);
+    }
+
+    this.initialized = false;
+  }
+
+  private void registerEventHandlers() {
+    eventBus.addHandler(ResourceChangedEvent.getType(), this);
+
+    subscribeToGitEvents.addSubscriber(this);
+  }
+
+  /** Invoked each time when panel is activated. */
+  @Override
+  public void onOpen() {
+    if (!initialized) {
+      loadPanelData();
+      registerEventHandlers();
+
+      initialized = true;
+    }
+  }
+
+  /** Queries from server all data needed to initialize the panel. */
+  private void loadPanelData() {
+    this.changes = new HashMap<>();
+
+    for (Project project : appContext.getProjects()) {
+      if (isProjectUnderGit(project)) {
+        view.addRepository(project.getName());
+        reloadRepositoryData(project);
+      }
+    }
+  }
+
+  private void reloadRepositoryData(Project project) {
+    service
+        .diff(project.getLocation(), null, NAME_STATUS, false, 0, REVISION, false)
+        .then(
+            diff -> {
+              MutableAlteredFiles alteredFiles = new MutableAlteredFiles(project, diff);
+              changes.put(project.getName(), alteredFiles);
+              view.updateRepositoryChanges(project.getName(), alteredFiles.getFilesQuantity());
+              // update changed files list if this repository is selected
+              if (project.getName().equals(view.getSelectedRepository())) {
+                updateChangedFiles(alteredFiles);
+              }
+            })
+        .catchError(
+            arg -> {
+              notificationManager.notify(locale.diffFailed(), FAIL, NOT_EMERGE_MODE);
+            });
+  }
+
+  @Override
+  public String getTitle() {
+    return locale.panelTitle();
+  }
+
+  @Override
+  public SVGResource getTitleImage() {
+    return gitResources.git();
+  }
+
+  @Override
+  public IsWidget getView() {
+    return view;
+  }
+
+  @Override
+  public String getTitleToolTip() {
+    return locale.panelTitleToolTip();
+  }
+
+  @Override
+  public void go(AcceptsOneWidget container) {
+    container.setWidget(view);
+  }
+
+  public void show() {
+    partStack.setActivePart(this);
+    onOpen();
+  }
+
+  public void hide() {
+    partStack.minimize();
+  }
+
+  public boolean isOpened() {
+    return partStack.getActivePart() == this;
+  }
+
+  @Override
+  public void onFileChanged(String endpointId, FileChangedEventDto dto) {
+    String projectName = extractProjectName(dto.getPath());
+    MutableAlteredFiles alteredFiles = changes.get(projectName);
+
+    switch (dto.getStatus()) {
+      case MODIFIED:
+        if (alteredFiles.addFile(removeProjectName(dto.getPath()), Status.MODIFIED)) {
+          view.updateRepositoryChanges(projectName, alteredFiles.getFilesQuantity());
```

Comment:
```
+          view.updateRepositoryChanges(projectName, alteredFiles.getFilesQuantity());
 +          if (projectName.equals(selectedProjectName)) {
 +            updateChangedFiles(alteredFiles);
 +          }
```
The common part can be extracted and processed under the `switch`

Reply:
Can you please elaborate you idea. I can move common code to another method, but still don't see how to process it under the switch block.